runtime/connections.go

```diff
@@ -290,29 +287,47 @@ func (r *Runtime) ConnectorConfig(ctx context.Context, instanceID, name string)
 	return res, nil
 }
 
-// ResolveConnectorProperties resolves templating in the provided connector's properties.
+// resolveConnectorProperties resolves templating in the provided connector's properties.
 // It always returns a clone of the properties, even if no templating is found, so the output is safe for further mutations.
-func ResolveConnectorProperties(environment string, vars map[string]string, c *runtimev1.Connector) (map[string]string, error) {
-	res := maps.Clone(c.Config)
+func resolveConnectorProperties(environment string, vars map[string]string, c *runtimev1.Connector) (map[string]any, error) {
+	res := c.Config.AsMap()
 	if res == nil {
-		res = make(map[string]string)
+		res = make(map[string]any)
+	}
+
+	td := parser.TemplateData{
+		Environment: environment,
+		Variables:   vars,
 	}
 
-	// Resolve templating in properties that use it
 	for _, k := range c.TemplatedProperties {
 		v, ok := res[k]
 		if !ok {
 			continue
 		}
-		v, err := parser.ResolveTemplate(v, parser.TemplateData{
-			Environment: environment,
-			Variables:   vars,
-		}, true)
+		// because we only added root value string properties in TemplatedProperties while parsing
+		s, ok := v.(string)
+		if !ok {
+			continue
+		}
+		v, err := parser.ResolveTemplate(s, td, true)
 		if err != nil {
 			return nil, fmt.Errorf("failed to resolve templated property %q: %w", k, err)
 		}
 		res[k] = v
 	}
+	// Resolve the remaining keys.
+	for k, v := range res {
+		// if value is root level string properties is must have already be resolved using TemplatedProperties
+		if _, ok := v.(string); !ok {
+			continue
+		}
+		v, err := parser.ResolveTemplateRecursively(v, td, true)
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve template: %w", err)
+		}
+		res[k] = v
+	}
```

Comment:
These `v.(string)` feel hacky and tightly coupled to the behavior of how `TemplatedProperties` are parsed. I would expect that templating is applied for any property in `TemplatedProperties`, and not for any other.

So instead of doing `v.(string)` checks here, consider if you can refactor the parser such that it also adds fields that have nested templating to `TemplatedProperties`.